aspnetcore/fundamentals/static-files.md

```diff
@@ -1,17 +1,1104 @@
 ---
 title: Static files in ASP.NET Core
 author: wadepickett
-description: Learn how to serve and secure static files and configure static file hosting middleware behaviors in an ASP.NET Core web app.
+description: Learn how to serve and secure static files and configure Map Static Assets endpoint conventions and Static Files Middleware in ASP.NET Core web apps.
 monikerRange: '>= aspnetcore-3.1'
 ms.author: wpickett
 ms.custom: mvc
-ms.date: 03/18/2025
+ms.date: 08/20/2025
 uid: fundamentals/static-files
 ---
 # Static files in ASP.NET Core
 
 [!INCLUDE[](~/includes/not-latest-version.md)]
 
-[!INCLUDE[](~/fundamentals/static-files/includes/static-files8.md)]
+Static files, also called static assets, are files an ASP.NET Core app that aren't dynamically generated and served directly to clients on request, such as HTML, CSS, image, and JavaScript files.
 
-[!INCLUDE[](~/fundamentals/static-files/includes/static-files6.md)]
+For Blazor static files guidance, which adds to or supersedes the guidance in this article, see <xref:blazor/fundamentals/static-files>.
+
+:::moniker range=">= aspnetcore-9.0"
+
+## Map Static Assets (`MapStaticAssets`)
+
+<xref:Microsoft.AspNetCore.Builder.StaticAssetsEndpointRouteBuilderExtensions.MapStaticAssets%2A> maps static files produced during the build as endpoints with the following features:
+
+* An asset is served once until the file changes or the browser clears its cache. The [`ETag`](https://developer.mozilla.org/docs/Web/HTTP/Headers/ETag) and [Last-Modified](https://developer.mozilla.org/docs/Web/HTTP/Headers/Last-Modified) headers are set.
+* The browser is prevented from using stale assets after an app is updated. The [`Last-Modified`](https://developer.mozilla.org/docs/Web/HTTP/Headers/Last-Modified) header is set.
+* The appropriate [caching headers](https://developer.mozilla.org/docs/Web/HTTP/Headers/Cache-Control) are set on the response.
+* [Caching Middleware](xref:performance/caching/middleware) is used.
```

Comment:
The Caching middleware isn't actually a feature of MapStaticAssets. It's a separate independent feature.

```suggestion
```